src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -1190,7 +1204,7 @@ export class SearchSource {
         buildExpressionFunction<EsdslExpressionFunctionDefinition>('esdsl', {
           size: body?.size,
           dsl: JSON.stringify({}),
-          index: index?.id,
+          index: typeof index === 'string' ? index : `${dataView?.id}`,
```

Comment:
Just one question here: if `dataView` is undefined the `index` value would be set to `"undefined"`. Is that ok?